Docs: yamllint rule

oraNod opened the pull request:
This PR updates documentation for the yaml rule.

ssbarnea commented:
You need to keep the `yaml.md` filename. We had to avoid using `yaml.py` on this rule, but the documentation must remain `yaml.md`. Is the only exception.

oraNod commented:
> You need to keep the `yaml.md` filename. We had to avoid using `yaml.py` on this rule, but the documentation must remain `yaml.md`. Is the only exception.

Thanks, Sorin. I wasn't sure about this one so took a punt and renamed the md file. I've changed it back.
